packages/package_info/lib/package_info.dart

```diff
@@ -17,35 +17,36 @@ const MethodChannel _kChannel =
 /// print("Version is: ${packageInfo.version}");
 /// ```
 class PackageInfo {
+  /// Constructs an instance with the given values for testing. [PackageInfo]
+  /// instances constructed this way won't actually reflect any real information
+  /// from the platform, just whatever was passed in at construction time.
+  ///
+  /// See [fromPlatform] for the right API to get a [PackageInfo] that's
+  /// actually populated with real data.
   PackageInfo({
     this.appName,
     this.packageName,
     this.version,
     this.buildNumber,
   });
 
-  static Future<PackageInfo> _fromPlatform;
+  static PackageInfo _fromPlatform;
 
   /// Retrieves package information from the platform.
   /// The result is cached.
   static Future<PackageInfo> fromPlatform() async {
-    if (_fromPlatform == null) {
-      final Completer<PackageInfo> completer = Completer<PackageInfo>();
-
-      _kChannel.invokeMapMethod<String, dynamic>('getAll').then(
-          (dynamic result) {
-        final Map<dynamic, dynamic> map = result;
-
-        completer.complete(PackageInfo(
-          appName: map["appName"],
-          packageName: map["packageName"],
-          version: map["version"],
-          buildNumber: map["buildNumber"],
-        ));
-      }, onError: completer.completeError);
-
-      _fromPlatform = completer.future;
```

Comment:
@mehmetf this looked like it was converting a Future into a callback and then back into a Future again for no real reason, and could just be simplified to awaiting the future directly. Am I missing something here? I want to make sure that there's not something extra happening here that I'm missing with my simplified refactor.

This was throwing an `unawaited_futures` lint from the `.then()` callback conversion and completer. If that is accomplishing something extra I could just comment the lint out on this line instead.

Reply:
I don't recall this doing anything special (e.g. completer or its future is not cached). Likely the change is safe.